src/commonControls/questionnaireControl/QuestionnaireControl.ts

```diff
@@ -1144,10 +1155,30 @@ export class QuestionnaireControl extends Control implements InteractionModelCon
         return;
     }
 
+    private wantsToKeepAplInitiative(input: ControlInput): boolean {
+        const content = this.getQuestionnaireContent(input);
+        try {
+            okIf(this.inputWasAnswerByTouch);
```

Comment:
more like a question: is it means any touch event will make this control be able to take Initiative ?

Reply:
this.inputWasAnswerByTouch will only be true if the input on this turn was specifically a touch event on a radio button of this control.  In this situation we will keep the APL up and not speak a prompt or open the mic.